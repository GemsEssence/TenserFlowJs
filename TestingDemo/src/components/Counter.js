import React from 'react';
import { View, Text, Button, StyleSheet } from 'react-native';

const Counter = ({ count, setCount }) => {
  return (
    <View
      style={styles.container}
      testID="counter-container"
      accessibilityLabel="counter-container"
    >
      <Text
        style={{ fontSize: 20 }}
        testID="counter-text"
        accessibilityLabel="counter-text"
      >
        Count: {count}
      </Text>

      <Button
        title="Increment"
        onPress={() => setCount(count + 1)}
        testID="increment-button"
        accessibilityLabel="increment-button"
      />

      <Button
        title="Decrement"
        onPress={() => setCount(count - 1)}
        testID="decrement-button"
        accessibilityLabel="decrement-button"
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default Counter;











// import React, { useState } from 'react';
// import { View, Text, Button, StyleSheet } from 'react-native';

// const Counter = () => {
//   const [count, setCount] = useState(0);

//   return (
//     <View
//       style={styles.container}
//       testID="counter-container"
//       accessibilityLabel="counter-container"
//     >
//       <Text style={{
//         fontSize: 20
//       }} testID="counter-text" accessibilityLabel="counter-text">
//         Count: {count}
//       </Text>

//       <Button
//         title="Increment"
//         onPress={() => setCount(count + 1)}
//         testID="increment-button"
//         accessibilityLabel="increment-button"
//       />

//       <Button
//         title="Decrement"
//         onPress={() => setCount(count - 1)}
//         testID="decrement-button"
//         accessibilityLabel="decrement-button"

//       />
//     </View>
//   );
// };

// const styles = StyleSheet.create({
//   container: {
//     flex: 1,
//     justifyContent: 'center',
//     alignItems: 'center',
//   },
// });

// export default Counter;
