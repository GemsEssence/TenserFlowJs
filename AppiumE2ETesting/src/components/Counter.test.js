import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Counter from './Counter';

describe('Counter Component', () => {
  it('renders initial count', () => {
    const { getByText } = render(<Counter />);
    expect(getByText('Count: 0')).toBeTruthy();
  });

  it('increments count when button is pressed', () => {
    const { getByText, getByTestId } = render(<Counter />);
    fireEvent.press(getByTestId('increment-button'));
    expect(getByText('Count: 1')).toBeTruthy();
  });

  it('decrements count when button is pressed', () => {
    const { getByText, getByTestId } = render(<Counter />);
    fireEvent.press(getByTestId('decrement-button'));
    expect(getByText('Count: -1')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const { toJSON } = render(<Counter />);
    expect(toJSON()).toMatchSnapshot();
  });
});