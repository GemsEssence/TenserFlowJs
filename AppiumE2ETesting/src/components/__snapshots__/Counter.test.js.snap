// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Counter Component matches snapshot 1`] = `
<View
  accessibilityLabel="counter-container"
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="counter-container"
>
  <Text
    accessibilityLabel="counter-text"
    testID="counter-text"
  >
    Count: 
    0
  </Text>
  <View
    accessibilityLabel="increment-button"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="increment-button"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Increment
      </Text>
    </View>
  </View>
  <View
    accessibilityLabel="decrement-button"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="decrement-button"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Decrement
      </Text>
    </View>
  </View>
</View>
`;
